Automattic/crowdsignal-ui
# Update project renderer: work async

This PR updates the project renderer app/workspace.

The new backend layout ( D69570-code ) now only prints some data- attributes on the project container. From there, the project renderer will get:

  - project ID/CODE (the new endpoints can handle both)
  - respondent ID, provided by backend, default to empty
  - page number, array index of the content.pages, defaults to 0 (zero)
  - startDate, timestamp 

A new rest-api method was added on the component, `fetchProjectPage` to do the initial request. While we still don't handle more than one page, subsequent pages should come from the response after POSTing the endpoint

## Test instructions

Create a project using the `dashboard` workspace. Save and publish. Copy the project's ID from the address bar.

Start the project renderer with `yarn workspace @crowdsignal/project-renderer start`, visit crowdsignal.localhost:9000 and append the copied project ID

The project should render as you published it.

If you remembered to add a submit button, you should be able to submit answers (although the project will not reload or anything)

## Update

Since a built version of this update is already deployed, you can copy the project's URL by clicking the "Share" button, paste it on the address bar and append `?use-project` to see it rendering in production
